FAILED: runGitDescribeWithMatchOption, has trouble on Travis

This test `runGitDescribeWithMatchOption` has trouble on travis, see: https://travis-ci.org/ktoso/maven-git-commit-id-plugin/builds/43299247

```
java.lang.RuntimeException: Failed to execute git command (`git symbolic-ref HEAD` @ /home/travis/build/ktoso/maven-git-commit-id-plugin/target/sandbox/my-pom-project/my-jar-module)!
    at pl.project13.maven.git.NativeGitProvider.runGitCommand(NativeGitProvider.java:271)
    at pl.project13.maven.git.NativeGitProvider.tryToRunGitCommand(NativeGitProvider.java:236)
    at pl.project13.maven.git.NativeGitProvider.getBranch(NativeGitProvider.java:76)
    at pl.project13.maven.git.NativeGitProvider.getBranchName(NativeGitProvider.java:72)
    at pl.project13.maven.git.GitDataProvider.determineBranchName(GitDataProvider.java:142)
    at pl.project13.maven.git.GitDataProvider.loadGitData(GitDataProvider.java:86)
    at pl.project13.maven.git.GitCommitIdMojo.loadGitDataWithNativeGit(GitCommitIdMojo.java:477)
    at pl.project13.maven.git.GitCommitIdMojo.loadGitData(GitCommitIdMojo.java:460)
    at pl.project13.maven.git.GitCommitIdMojo.execute(GitCommitIdMojo.java:320)
    at pl.project13.maven.git.GitCommitIdMojoIntegrationTest.runGitDescribeWithMatchOption(GitCommitIdMojoIntegrationTest.java:649)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:606)
    at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:45)
    at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:15)
    at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:42)
    at junitparams.internal.InvokeParameterisedMethod.evaluate(InvokeParameterisedMethod.java:184)
    at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:28)
    at junitparams.internal.ParameterisedTestMethodRunner.runMethodInvoker(ParameterisedTestMethodRunner.java:48)
    at junitparams.internal.ParameterisedTestMethodRunner.runTestMethod(ParameterisedTestMethodRunner.java:42)
    at junitparams.internal.ParameterisedTestClassRunner.runParameterisedTest(ParameterisedTestClassRunner.java:143)
    at junitparams.JUnitParamsRunner.runChild(JUnitParamsRunner.java:405)
    at junitparams.JUnitParamsRunner.runChild(JUnitParamsRunner.java:383)
    at org.junit.runners.ParentRunner$3.run(ParentRunner.java:231)
    at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:60)
    at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:229)
    at org.junit.runners.ParentRunner.access$000(ParentRunner.java:50)
    at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:222)
    at org.junit.runners.ParentRunner.run(ParentRunner.java:300)
    at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:252)
    at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:141)
    at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:112)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:606)
    at org.apache.maven.surefire.util.ReflectionUtils.invokeMethodWithArray(ReflectionUtils.java:189)
    at org.apache.maven.surefire.booter.ProviderFactory$ProviderProxy.invoke(ProviderFactory.java:165)
    at org.apache.maven.surefire.booter.ProviderFactory.invokeProvider(ProviderFactory.java:85)
    at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:115)
    at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:75)
Caused by: java.io.IOException: Git command exited with invalid status [128]: stdout: ``, stderr: `fatal: ref HEAD is not a symbolic ref`
    at pl.project13.maven.git.NativeGitProvider$JavaProcessRunner.run(NativeGitProvider.java:315)
    at pl.project13.maven.git.NativeGitProvider.runGitCommand(NativeGitProvider.java:267)
    ... 41 more
TEST: Will prepare sandbox repository based on: [src/test/resources/_git_one_commit]
TEST: Will prepare sandbox repository based on: [src/test/resources/_git_one_commit]
TEST: Will prepare sandbox repository based on: [src/test/resources/_git_one_commit]
TEST: Will prepare sandbox repository based on: [src/test/resources/_git_one_commit]
TEST: Will prepare sandbox repository based on: [src/test/resources/_git_one_commit]
TEST: Will prepare sandbox repository based on: [src/test/resources/_git_one_commit]
Tests run: 50, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 3.04 sec <<< FAILURE!
[1] true (runGitDescribeWithMatchOption)(pl.project13.maven.git.GitCommitIdMojoIntegrationTest)  Time elapsed: 0.037 sec  <<< FAILURE!
```

--- a/src/main/java/pl/project13/maven/git/NativeGitProvider.java
+++ b/src/main/java/pl/project13/maven/git/NativeGitProvider.java
@@ -73,9 +73,15 @@ public class NativeGitProvider extends GitDataProvider {
   }
 
   private String getBranch(File canonical) {
-    String branch = tryToRunGitCommand(canonical, "symbolic-ref HEAD");
-    if (branch != null) {
-      branch = branch.replace("refs/heads/", "");
+    String branch = null;
+    try{
+      branch = tryToRunGitCommand(canonical, "symbolic-ref HEAD");
+      if (branch != null) {
+        branch = branch.replace("refs/heads/", "");
+      }
+    }catch(RuntimeException e){
+      // it seems that git repro is in 'DETACHED HEAD'-State, using Commid-Id as Branch
+      branch = getCommitId();
     }
     return branch;
   }

--- a/src/test/java/pl/project13/maven/git/GitCommitIdMojoIntegrationTest.java
+++ b/src/test/java/pl/project13/maven/git/GitCommitIdMojoIntegrationTest.java
@@ -621,11 +621,6 @@ public class GitCommitIdMojoIntegrationTest extends GitIntegrationTest {
   @Test
   @Parameters(method = "useNativeGit")
   public void runGitDescribeWithMatchOption(boolean useNativeGit) throws Exception {
-    if (System.getenv().get("TRAVIS").equals("true") && System.getenv().get("CI").equals("true")) {
-      // FIXME: this test has trouble on travis, I think it's because of not pulling everything?
-      return;
-    }
-
     // given
     mavenSandbox.withParentProject("my-pom-project", "pom")
                 .withChildProject("my-jar-module", "jar")

